src/runtime-rs/crates/runtimes/src/manager.rs

```diff
@@ -0,0 +1,247 @@
+                    Ok(Response::ShutdownContainer)
+                }
+                Request::WaitProcess(process_id) => {
+                    let exit_status = cm.wait_process(&process_id).await.context("wait process")?;
+                    if cm.is_sandbox_container(&process_id).await {
+                        sandbox.stop().await.context("stop sandbox")?;
+                    }
+                    Ok(Response::WaitProcess(exit_status))
+                }
+                Request::StartProcess(process_id) => {
+                    let shim_pid = cm
+                        .start_process(&process_id)
+                        .await
+                        .context("start process")?;
+                    Ok(Response::StartProcess(shim_pid))
+                }
+
+                Request::StateProcess(process_id) => {
+                    let state = cm
+                        .state_process(&process_id)
+                        .await
+                        .context("state process")?;
+                    Ok(Response::StateProcess(state))
+                }
+                Request::PauseContainer(container_id) => {
+                    cm.pause_container(&container_id)
+                        .await
+                        .context("pause container")?;
+                    Ok(Response::PauseContainer)
+                }
+                Request::ResumeContainer(container_id) => {
+                    cm.resume_container(&container_id)
+                        .await
+                        .context("resume container")?;
+                    Ok(Response::ResumeContainer)
+                }
+                Request::ResizeProcessPTY(req) => {
+                    cm.resize_process_pty(&req).await.context("resize pty")?;
+                    Ok(Response::ResizeProcessPTY)
+                }
+                Request::StatsContainer(container_id) => {
+                    let stats = cm
+                        .stats_container(&container_id)
+                        .await
+                        .context("stats container")?;
+                    Ok(Response::StatsContainer(stats))
+                }
+                Request::UpdateContainer(req) => {
+                    cm.update_container(req).await.context("update container")?;
+                    Ok(Response::UpdateContainer)
+                }
+                Request::Pid => Ok(Response::Pid(cm.pid().await.context("pid")?)),
+                Request::ConnectContainer(container_id) => Ok(Response::ConnectContainer(
+                    cm.connect_container(&container_id)
+                        .await
+                        .context("connect")?,
+                )),
+            }
+        }
+    }
+}
+
+/// Config override ordering(high to low):
+/// 1. podsandbox annotation
+/// 2. shimv2 create task option
+/// TODO: https://github.com/kata-containers/kata-containers/issues/3961
+/// 3. environment
+fn load_config(spec: &oci::Spec) -> Result<TomlConfig> {
+    // TODO:
```

Comment:
?

Reply:
Thanks for your review, removed line 234.